assets/js/components/settings/SettingsAdmin.js

```diff
@@ -24,24 +24,50 @@ import { __ } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import { MODULES_SEARCH_CONSOLE } from '../../modules/search-console/datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';
 import Layout from '../layout/Layout';
 import { Grid, Cell, Row } from '../../material-components';
 import OptIn from '../OptIn';
 import ResetButton from '../ResetButton';
 import { useFeature } from '../../hooks/useFeature';
 import SettingsCardKeyMetrics from './SettingsCardKeyMetrics';
 import SettingsPlugin from './SettingsPlugin';
+const { useSelect } = Data;
 
 export default function SettingsAdmin() {
 	const userInputEnabled = useFeature( 'userInput' );
 
+	const searchConsoleModuleConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'search-console' )
+	);
+	const analyticsModuleConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
+	);
+	const searchConsoleIsGatheringData = useSelect(
+		( select ) =>
+			searchConsoleModuleConnected &&
+			select( MODULES_SEARCH_CONSOLE ).isGatheringData()
+	);
+	const analyticsIsGatheringData = useSelect(
+		( select ) =>
+			analyticsModuleConnected &&
+			select( MODULES_ANALYTICS_4 ).isGatheringData()
+	);
```

Comment:
Good point, I've resolved this alongside rolling the values into a single `useSelect()` as per your comment below, and also removed the request interception.

Reply:
Hmm. I now realise that a similar point applies to the previous [PR](https://github.com/google/site-kit-wp/pull/7198/files) for this issue, where checks for gathering data state have been added to `KeyMetricsSetupCTAWidget` without a short circuit check for the `userInput` feature flag, and tests fixed where they needn't be (yet, until we remove the flag of course!).

Looking into it a bit further, it appears we could simplify matters by making the widget registration itself dependent on the feature flag, like they other KM widgets, and remove the check for the flag entirely from `KeyMetricsSetupCTAWidget`

I'd be happy to make this change and revert those test updates as part of this PR if you don't think it would be overly complicating things. Alternatively as this issue has already gone way over estimate we could address this in a followup issue (or simply leave it I suppose as the changes will actually be relevant once the feature flag is removed). What do you think?

Reply:
Thanks for the additional thoughts here. We should not be checking for a feature flag within a component that itself should only be rendered if that feature is active (`KeyMetricsSetupCTAWidget`). We've seen this in a few places and need to make sure we're always guarding the rendering of these feature-specific components outside of them.

In the past, we had always conditionally registered widgets for features so this would be consistent with that. However, in this case the entire widget context and area are being conditionally rendered already, so this isn't really necessary but the conditional registration would be consistent with what we've done in the past.

> Alternatively as this issue has already gone way over estimate we could address this in a followup issue (or simply leave it I suppose as the changes will actually be relevant once the feature flag is removed). What do you think?

Thanks for flagging. We can address in a follow-up as to not balloon the scope here, but there is a risk to leaving the test changes in place (particularly request interception) because it would mask any subsequent lack of feature flag checks and allow these kinds of side effects to potentially creep into "production" where they shouldn't be yet. I'm happy to save as much as we can to save time later when we do remove the feature flag but we want our tests to reflect the pre-feature state to guard against this kind of leak.

Reply:
Thanks for pointing out that the Key Metrics context isn't actually rendered when the feature flag isn't set. As discussed on Slack it turns out I rushed to judgement a bit on my comment above and the PR changes are OK - the `KeyMetricsSetupCTAWidget` _is_ only rendered when the `userInput` feature flag is enabled (thanks to the conditional context rendering), and the related test suite does enable the flag and tests user input related scenarios.

So, in fact the only thing needing to be addressed in `KeyMetricsSetupCTAWidget` is the redundant check for the `userInput` flag...